tests/ui/lint/non-local-defs/suggest-moving-inner.stderr

```diff
@@ -0,0 +1,31 @@
+warning: non-local `impl` definition, `impl` blocks should be written at the same level as their item
+  --> $DIR/suggest-moving-inner.rs:12:5
+   |
+LL |     impl<T> Trait<InsideMain> for &Vec<below::Type<(InsideMain, T)>>
+   |     ^^^^^^^^-----^^^^^^^^^^^^^^^^^----------------------------------
+   |             |                     |
+   |             |                     `&'_ Vec<below::Type<(InsideMain, T)>>` is not local
+   |             `Trait` is not local
+   |
+   = note: `impl` may be usable in bounds, etc. from outside the expression, which might e.g. make something constructible that previously wasn't, because it's still on a publicly-visible type
+   = note: an `impl` is never scoped, even when it is nested inside an item, as it may impact type checking outside of that item, which can be the case if neither the trait or the self type are at the same nesting level as the `impl`
+help: move this `impl` block outside of the current function `main`
+  --> $DIR/suggest-moving-inner.rs:12:5
+   |
+LL |       impl<T> Trait<InsideMain> for &Vec<below::Type<(InsideMain, T)>>
+   |       ^             ----------           -----------  ---------- may need to be moved as well
```

Comment:
I see that you're hedging your bets that the type itself might also be in a weird scope that would otherwise not be accessible to the `impl` in the new location. Ideally we would check that that is the case before saying anything.

Reply:
I don't understand your comment, it's not betting, we are checking in the code if the type is at the same nesting, and if it is we emit the note saying that it might (as in may not be something you might want to do) need to be moved, there is no guess here.

Reply:
Ah! I see. In that case I would make it a message explaining alternative ways of sealing traits and types?

Reply:
It doesn't make sense to recommend sealing the trait since if we warn that means it already not local so moving the `impl` shouldn't change it's availability/visibility.

Regarding sealing the type, idk if that's a thing, but it also wouldn't change anything, since if it is not local it is already leaked.

In general it is very difficult to say how users should change their code to make a impl local (which from the type-system point-of-view doesn't exist anyway), so in general the lint "fallback" to proposing to make the impl def true to what it is, a global impl, and as such should be in a "global context" (module) and not inside a function, consts, array count, ...